underfs/b2/src/main/java/alluxio/underfs/b2/B2UnderFileSystem.java

```diff
@@ -0,0 +1,374 @@
+            USER_AGENT);
+
+    try {
+      B2AccountAuthorization accountAuthorization = client.getAccountAuthorization();
+      LOG.debug("B2AccountAuthorization {}", accountAuthorization);
+
+      B2Allowed allowed = accountAuthorization.getAllowed();
+      LOG.debug("B2Allowed {}", allowed);
+
+      String accountId = accountAuthorization.getAccountId();
+      List<String> capabilities = allowed.getCapabilities();
+
+      String bucketNameFromURI = UnderFileSystemUtils.getBucketName(alluxioURI);
+      LOG.debug("bucket name from uri {}", bucketNameFromURI);
+
+      String bucketId;
+      String bucketName;
+      if (allowed.getBucketId() != null) {
+        bucketId = allowed.getBucketId();
+        bucketName = allowed.getBucketName();
+      } else {
+        B2Bucket bucket = client.getBucketOrNullByName(bucketNameFromURI);
+        LOG.debug("B2Bucket {}", bucket);
+        bucketId = bucket.getBucketId();
+        bucketName = bucket.getBucketName();
+      }
+
+      short bucketMode = B2Utils.translateBucketAcl(capabilities);
+
+      return new B2UnderFileSystem(alluxioURI, conf, client, bucketId, bucketName, bucketMode,
+          accountId);
+    } catch (B2Exception | NullPointerException e) {
+      LOG.error("Failed to instantiate B2UnderFileSystem client with key:{} and URI: {}",
+          conf.get(PropertyKey.B2_ACCESS_KEY), alluxioURI);
+      Throwables.propagateIfPossible(e, ServiceException.class);
+    }
+    return null;
+  }
+
+  @Override
+  protected OutputStream createObject(String key) throws IOException {
+    return new B2OutputStream(mBucketId, key, mB2StorageClient,
+        mUfsConf.getList(PropertyKey.TMP_DIRS, ","));
+  }
+
+  @Override
+  protected boolean deleteObject(String key) throws IOException {
+    try {
+      B2ListFileVersionsRequest request =
+          B2ListFileVersionsRequest.builder(mBucketId).setStartFileName(key).setPrefix(key).build();
+      for (B2FileVersion version : mB2StorageClient.fileVersions(request)) {
+        if (version.getFileName().equals(key)) {
+          mB2StorageClient.deleteFileVersion(version);
+        } else {
+          break;
+        }
+      }
+    } catch (B2Exception e) {
+      LOG.error("Failed to delete {}", key, e);
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  protected String getFolderSuffix() {
+    LOG.debug("getFolderSuffix {}", FOLDER_SUFFIX);
+    return FOLDER_SUFFIX;
+  }
+
+  @Nullable
+  @Override
+  protected ObjectListingChunk getObjectListingChunk(String key, boolean recursive)
+      throws IOException {
+    LOG.debug("getObjectListingChunk({}, {})", key, recursive);
+    String delimiter = recursive ? null : PATH_SEPARATOR;
+    key = PathUtils.normalizePath(key, PATH_SEPARATOR);
+    // In case key is root (empty string) do not normalize prefix.
+    key = key.equals(PATH_SEPARATOR) ? "" : key;
+
+    LOG.debug("key: {}", key);
+    LOG.debug("delimiter: {}", delimiter);
```

Comment:
Can we combine all the debug statements into a single log line with all the desired parameters?